src/components/TodoList.tsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import classNames from 'classnames';
+import { TodoItem } from '../components/TodoItem';
+import { Todo } from '../types/Todo';
+import { useState } from 'react';
+import React from 'react';
+
+type Props = {
+  filteredTodos: Todo[];
+  tempTodo: Todo | null;
+  handleDeleteTodo: (a: number) => Promise<void>;
+  handleUpdateTodo: (a: number, b: string, c: boolean) => Promise<void>;
+  loadingIds: number[];
+};
+
+export const TodoList: React.FC<Props> = ({
+  filteredTodos,
+  tempTodo,
+  handleDeleteTodo,
+  loadingIds,
+  handleUpdateTodo,
+}) => {
+  const [isEditing, setIsEditing] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState('');
+
+  const handleRenaimTodo = (id: number, title: string) => {
+    setIsEditing(id);
+    setNewTitle(title);
+  };
+
+  const handleNewTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const handleKeyDown = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+    id: number,
+    completed: boolean,
+  ) => {
+    if (event.key === 'Enter') {
+      setIsEditing(null);
+      if (newTitle.length === 0) {
+        handleDeleteTodo(id).catch(() => setIsEditing(id));
+      } else {
+        handleUpdateTodo(id, newTitle, completed).catch(() => setIsEditing(id));
+      }
+    }
+
+    if (event.key === 'Escape') {
+      setIsEditing(null);
+    }
+  };
+
+  const handleBlur = (id: number, completed: boolean) => {
+    setIsEditing(null);
+    if (newTitle.length === 0) {
```

Comment:
```suggestion
    if (!!newTitle.length) {
```

can be a bit simplify